lib/errors.ts

```
@@ -0,0 +1,88 @@
+): T {
+  if (error instanceof Error) {
+    // For custom errors that need specific arguments like filePath, we'll use the default code
+    return new ErrorClass(message, 'ERROR', error);
+  }
+  return new ErrorClass(message + ': ' + String(error), 'ERROR');
+}
+
+/**
+ * Check if an error is of a specific custom error type
+ */
+export function isErrorOfType<T extends Error>(error: unknown, errorType: new (...args: any[]) => T): error is T {
```

Biome (1.9.4)

[error] Unexpected any. Specify a different type.

any disables many type checking rules. Its use should be avoided.

(lint/suspicious/noExplicitAny)